src/core/collision.cpp

```diff
@@ -579,18 +605,38 @@ void handle_collisions() {
       Particle *const p1 = local_particles[c.pp1];
       Particle *const p2 = local_particles[c.pp2];
 
-      // If we have none of the two partic;es, only increase the counter for the
-      // next id to use
-      if (p1 == NULL and p2 == NULL) {
+      // Only nodes take part in particle creation and binding
+      // that see both particles
+
+      // If we have
+      // * none of the two partic;es or
+      // * both are ghosts,
+      // * or we have only one and it is a ghost
+      // we only increase the counter for the ext id to use based on the
+      // number of particles created by other nodes
+      if ((!p1 and !p2) ||
+          ((p1 && p1->l.ghost) && (p2 && p2->l.ghost)) ||
+          (!p1 && p2->l.ghost) || (!p2 && p1->l.ghost)) {
+        // Increase local counters
         if (collision_params.mode & COLLISION_MODE_VS) {
           added_particle(current_vs_pid);
           current_vs_pid++;
         }
         // For glue to surface, we have only one vs
         added_particle(current_vs_pid);
         current_vs_pid++;
+        if (collision_params.mode == COLLISION_MODE_GLUE_TO_SURF) {
+          if (p1)
+            if (p1->p.type == collision_params.part_type_to_be_glued) {
+              p1->p.type = collision_params.part_type_after_glueing;
+            }
+          if (p2)
+            if (p2->p.type == collision_params.part_type_to_be_glued) {
+              p2->p.type = collision_params.part_type_after_glueing;
+            }
+        }
 
-      } else { // We have at least one
+      } else { // We consider teh pair
```

Comment:
Typo: "the". We could add here: "because at least one of p1 and p2 is local to this node".